samples/csharp/getting-started/Regression_AutoML/TaxiFarePrediction/Program.cs

```diff
@@ -0,0 +1,329 @@
+                    Console.WriteLine($"Predicted : {FarePrediction.FareAmount}");
+                    Console.WriteLine($"Actual:    {testData[i].FareAmount}");
+                    Console.WriteLine($"-------------------------------------------------");
+                }
+
+                // Regression Line calculation explanation:
+                // https://www.khanacademy.org/math/statistics-probability/describing-relationships-quantitative-data/more-on-regression/v/regression-line-example
+
+                double minY = yTotal / totalNumber;
+                double minX = xTotal / totalNumber;
+                double minXY = xyMultiTotal / totalNumber;
+                double minXsquare = xSquareTotal / totalNumber;
+
+                double m = ((minX * minY) - minXY) / ((minX * minX) - minXsquare);
+
+                double b = minY - (m * minX);
+
+                //Generic function for Y for the regression line
+                // y = (m * x) + b;
+
+                double x1 = 1;
+                //Function for Y1 in the line
+                double y1 = (m * x1) + b;
+
+                double x2 = 39;
+                //Function for Y2 in the line
+                double y2 = (m * x2) + b;
+
+                var xArray = new double[2];
+                var yArray = new double[2];
+                xArray[0] = x1;
+                yArray[0] = y1;
+                xArray[1] = x2;
+                yArray[1] = y2;
+
+                pl.col0(4);
+                pl.line(xArray, yArray);
+
+                // end page (writes output to disk)
+                pl.eop();
+
+                // output version of PLplot
+                pl.gver(out var verText);
+                Console.WriteLine("PLplot version " + verText);
+
+            } // the pl object is disposed here
+
+            // Open Chart File In Microsoft Photos App (Or default app, like browser for .svg)
+
+            Console.WriteLine("Showing chart...");
+            var p = new Process();
+            string chartFileNamePath = @".\" + chartFileName;
+            p.StartInfo = new ProcessStartInfo(chartFileNamePath)
+            {
+                UseShellExecute = true
+            };
+            p.Start();
+        }
+
+        public static string GetAbsolutePath(string relativePath)
+        {
+            FileInfo _dataRoot = new FileInfo(typeof(Program).Assembly.Location);
+            string assemblyFolderPath = _dataRoot.Directory.FullName;
+
+            string fullPath = Path.Combine(assemblyFolderPath, relativePath);
+
+            return fullPath;
+        }
+
+        /// <summary>
+        /// Print top models from AutoML experiment.
+        /// </summary>
+        private static void PrintTopModels(ExperimentResult<RegressionMetrics> experimentResult)
+        {
+            // Get top few runs ranked by R-Squared
+            var topRuns = experimentResult.RunDetails.OrderByDescending(r => r.ValidationMetrics.RSquared).Take(3);
```

Comment:
Does this sort correctly if nulls or Double.NaN are in the list?

Reply:
no! let me fix!